File: 프로그래머스/1/12912. 두 정수 사이의 합/두 정수 사이의 합.py
def solution(a, b):
    answer = 0

    if a > b:
        a, b = b, a

    if (b - a) % 2 == 0:
        answer += (a + b) * (b - a) / 2 + (a + b) / 2
    else:
        answer += (a + b) * (b - a + 1) / 2
    
    return answer

File: 프로그래머스/1/12912. 두 정수 사이의 합/두 정수 사이의 합.java
class Solution {
    public long solution(int a, int b) {
        long answer = 0;
        if (a > b) { // a <= b 유지
            int tmp = b;
            b = a;
            a = tmp;
        }
        
        long sum = (long)a + (long)b;
        long cnt = (long)b - (long)a;
        
        if ((b - a) % 2 == 0) { // a ~ b 사이 정수의 개수가 홀수
            answer += sum * cnt / 2 + sum / 2;
        }
        else { // a ~ b 사이 정수의 개수가 짝수
            answer += sum * (cnt + 1) / 2;
        }
        return answer;
    }
}